erpnext/stock/doctype/item/item.py

```diff
@@ -492,18 +492,26 @@ def set_shopping_cart_data(self, context):
 		context.shopping_cart = get_product_info_for_website(self.name, skip_quotation_creation=True)
 
 	def add_default_uom_in_conversion_factor_table(self):
-		uom_conv_list = [d.uom for d in self.get("uoms")]
-		if self.stock_uom not in uom_conv_list:
-			ch = self.append('uoms', {})
-			ch.uom = self.stock_uom
-			ch.conversion_factor = 1
-
-		to_remove = []
-		for d in self.get("uoms"):
-			if d.conversion_factor == 1 and d.uom != self.stock_uom:
-				to_remove.append(d)
-
-		[self.remove(d) for d in to_remove]
+		uoms_list = []
+		if not self.is_new():
+			if self.stock_uom != frappe.get_value("Item", self.name, "stock_uom"):
+				idx = 1
+				for d in self.get("uoms"):
+					if d.conversion_factor == 1 and d.uom != self.stock_uom:
+						self.remove(d)
+					else:
+						d.idx = idx
+						idx += 1
```

Comment:
1. mutating list while iterating over it will lead to runtime errors.
2. it's simpler to just reset the index instead of this complicated logic. Once all modifications are done...

```python
for idx, row in enumerate(doc.child_table, start=1):
    row.idx = idx
```